src/docfx/build/redirection/RedirectionProvider.cs

```diff
@@ -41,16 +41,16 @@ internal class RedirectionProvider
             {
                 var redirections = LoadRedirectionModel(errors, docsetPath, repository);
                 _redirectUrls = GetRedirectUrls(redirections, hostName);
+                _publishUrlMap = publishUrlMap;
                 _history =
                    new Lazy<(IReadOnlyDictionary<FilePath, FilePath> renameHistory, IReadOnlyDictionary<FilePath, (FilePath, SourceInfo?)> redirectionHistory)>(
                         () => GetRenameAndRedirectionHistory(redirections, _redirectUrls));
-                _publishUrlMap = publishUrlMap;
             }
         }
 
         public bool Contains(FilePath file)
         {
-            return _redirectUrls.ContainsKey(file);
+            return _redirectUrls.Keys.Select(x => x.Path).Contains(file.Path);
```

Comment:
consider create another dictionary using `PathString` as key to avoid this linear search? The redirection list is big.

Reply:
u r right!